### ISSUE TITLE
Some Noise points scattered in 3D space despite of both Lidar Data.

### ISSUE THREAD

**Issue by rahulsharma1234** [SeoDU/multi_fastlio#3]
Hi.

I am using MID360 and AVIA lidar combination.
MID is a bit tilted towards AVIA to get some decent overlapping of both lidar data.

My question is can i use MID AVIA combination for the same? or MID360 with MID360?

Observance:
I am doubting the laser output of 1st lidar is colliding sometime with other lidar laser and creating some random noisy points in any 3D space.

Prompt reply will be very helpful for us to decide which 2 lidar combination we can go ahead.
Attached images are for reference.

![Screenshot from 2024-02-19 17-02-11](https://github.com/SeoDU/multi_fastlio/assets/32838419/f47e02e9-14a4-44f9-98b4-ebdd7976cbaa)
![Screenshot from 2024-02-19 17-02-21](https://github.com/SeoDU/multi_fastlio/assets/32838419/641ca498-46f0-4c5a-9b8c-8d69d1d25204)

**Comment by hr2894235132** [comment]
> Hi.
> 
> I am using MID360 and AVIA lidar combination. MID is a bit tilted towards AVIA to get some decent overlapping of both lidar data.
> 
> My question is can i use MID AVIA combination for the same? or MID360 with MID360?
> 
> Observance: I am doubting the laser output of 1st lidar is colliding sometime with other lidar laser and creating some random noisy points in any 3D space.
> 
> Prompt reply will be very helpful for us to decide which 2 lidar combination we can go ahead. Attached images are for reference.
> 
> ![Screenshot from 2024-02-19 17-02-11](https://private-user-images.githubusercontent.com/32838419/305936635-f47e02e9-14a4-44f9-98b4-ebdd7976cbaa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMDAyNzMsIm5iZiI6MTcxMDI5OTk3MywicGF0aCI6Ii8zMjgzODQxOS8zMDU5MzY2MzUtZjQ3ZTAyZTktMTRhNC00NGY5LTk4YjQtZWJkZDc5NzZjYmFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDAzMTkzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhZmFkYmUxNDdjMzJjYmZmZjdlZWQzNGFiZTNlY2UwZDcxMjg3OWVmZjRlYjRjZWY5OTY0ZTEwNmU3YTA3ZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AECFFM9CePhu3TFyWH7mbvMOCxczgeKKrWSJ3L7aMAM) ![Screenshot from 2024-02-19 17-02-21](https://private-user-images.githubusercontent.com/32838419/305936648-641ca498-46f0-4c5a-9b8c-8d69d1d25204.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMDAyNzMsIm5iZiI6MTcxMDI5OTk3MywicGF0aCI6Ii8zMjgzODQxOS8zMDU5MzY2NDgtNjQxY2E0OTgtNDZmMC00YzVhLTliOGMtOGQ2OWQxZDI1MjA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDAzMTkzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkMmQ0MGYwNDUwNGFjZWMzNjRiNDhmNTNjNjMwMGZmYmIxZjJhMWIzNGMzZjAzMzRkNTRkN2M3MTM3ODg1ZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dDx9z1q5EVY3kaIeDl682hfgK0ECx3ANJUlzg2KqKiE)

Hello, I am using this error: [ERROR] [1710299899.926192655]: lidar loop back, clear buffer, 
have you encountered it?

**Comment by rahulsharma1234** [comment]
Hi @hr2894235132 ,
You will get that error if you are using in same machine without IP and PORT change of lidars.
I am able to get it done via docker.

link for reference:
https://github.com/rahulsharma11/multiple_livox_lidar_connection_docker

Hope this helps you.

**Comment by hr2894235132** [comment]
> Hi @hr2894235132 , You will get that error if you are using in same machine without IP and PORT change of lidars. I am able to get it done via docker.
> 
> link for reference: https://github.com/rahulsharma11/multiple_livox_lidar_connection_docker
> 
> Hope this helps you.

It should be like this, in fact I use two avia, these two avias are connected to the switch separately, and then connected to my computer through a network cable, one is 192.168.1.147, the other is 192.168.1.148. And my computer's ip is 192.168.1.102. Is that ok?
![image](https://github.com/SeoDU/multi_fastlio/assets/86639226/e3ff6152-0ae7-4f44-92fe-35d50f0c56a1)
![image](https://github.com/SeoDU/multi_fastlio/assets/86639226/fb263578-5a17-417b-9c65-f8a98aaa86b9)
![image](https://github.com/SeoDU/multi_fastlio/assets/86639226/4716763d-1f74-40bd-a43f-48c61e4eca2e)

**Comment by rahulsharma1234** [comment]
I don't think it will work although i have not tried that. It can give you buffer related error and same node register related error.
As i suggested earlier docker is good option.

**Comment by hr2894235132** [comment]
> I don't think it will work although i have not tried that. It can give you buffer related error and same node register related error. As i suggested earlier docker is good option.

Okay, thanks for your suggestion!  I'll go try the docker you suggested.
